Realm/RealmSwift/Object.swift

```diff
@@ -19,8 +19,48 @@
 import Realm
 import Realm.Private
 
+/**
+
+In Realm you define your model classes by subclassing RLMObject and adding properties to be persisted.
+You then instantiate and use your custom subclasses instead of using the RLMObject class directly.
+
+```swift
+class Dog: Object {
+    dynamic var name: String = ""
+    dynamic var adopted: Bool = false
+    let siblings = List<Dog>
+}
+```
+
+### Supported property types
+
+- `String`
+- `Int`, `Float`, `NSInteger`, `CGFloat`, `int`, `long`, `float`, and `double`
+- `Bool`, `BOOL`, and `bool`
+- `NSDate`
+- `NSData`
+- `Object` subclasses, so you can have many-to-one relationships.
+- `List<T: Object>`, so you can have many-to-many relationships.
+
+### Querying
+
+You can gets `Results` of an Object subclass via tha `objects(_:)` free function or
+`Realm.
+
+### Relationships
+
+See our [Cocoa guide](http://realm.io/docs/cocoa/latest) for more details.
+*/
 public class Object : RLMObjectBase, Equatable {
     // Get the names of all properties in the object which are of type List<>
+
+    /**
+    Returns the names of all properties in `obj` that are of type `List<>`.
+
+    :param: obj The object to be reflected.
+
+    :returns: Returns the names of all properties in `obj` that are of type `List<>`.
+    */
     public class func getGenericListPropertyNames(obj: AnyObject) -> NSArray {
```

Comment:
We should call this function `PRIVATE_getGenericListPropertyNames()` and have an all-caps `PRIVATE` disclaimer in the doc comment. It should be private but that isn't possible.